### ISSUE TITLE
Graphite v1.1 Datasource issues

### ISSUE THREAD

**Issue by shanson7** [grafana/grafana#10365]
- What Grafana version are you using? 
5.0
- What datasource are you using?
 Graphite
- What OS are you running grafana on? 
Ubuntu
- What did you do? 
Created a new panel and clicked "Select metric". Typed "host" to filter down the results.
- What was the expected result?
Upon initial click, I expected some suggested nodes / tag keys to appear. Upon typing, I expected filtered results to show up (particularly the tag key "host").
- What happened instead?
The initial list was only populated with all the Node 0's in the system. No tag keys were shown. Upon typing, "host" only Node 0's with "host" in the name showed up (and not even all of them).

- Requests/responses:
Initial request for find was:
 `api/datasources/proxy/25/metrics/find?from=1514374299&query=*&until=1514396019` which resulted in 11,960 initial nodes for tree traversal.
Initial request for tags autocomplete was:
`api/datasources/proxy/25/tags/autoComplete/tags` which resulted in 100 (max limit) sorted lexicographically, and did not include "host".

Upon typing `host` into the search box the exact same queries are run (timestamps slightly different). No filter to the request is added to filter down the results on the server side, which would allow "host" to show up.

To sum up:
1. A lot of initial nodes existing causes tag keys to be hidden in initial autocomplete (minor annoyance).
2. Adding text to the search does not propagate to the server, causing some items to remain hidden (Forces toggling to plain text input to continue).

**Comment by DanCech** [comment]
Fixed in #10139 (see dancech/grafana@f809477 & dancech/grafana@dfcdd85)

I separated the find change from the tag change because right now graphite will always answer from cache, but sending the prefix will result in a different find each time that won't be cached.

**Comment by bergquist** [comment]
Closing since its been fixed
